.github/workflows/rust.yaml

```diff
@@ -0,0 +1,53 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: Rust
+
+jobs:
+  tests-stable:
+    name: Tests (Stable)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        with:
+          cache-on-failure: true
+
+      - name: cargo test
+        run: cargo test --all --all-features
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          components: rustfmt, clippy
+          override: true
+
+      - uses: Swatinem/rust-cache@v1
+        with:
+          cache-on-failure: true
+
+      - name: cargo fmt
+        run: cargo +nightly fmt --all -- --check
+
+      - name: cargo clippy
+        run: cargo +nightly clippy --all --all-features -- -D warnings
```

Comment:
Add the missing newline.

Reply:
what do you mean by this?

Reply:
i believe you were missing an eol and gits needs it. we should add it to the options of the repo actually to add it automatically.